### ISSUE TITLE
Reducing the time to import sglang

### ISSUE THREAD

**Issue by merrymercy** [sgl-project/sglang#10492]
Now importing sglang takes too long. On my H100 machine, it took 12s to just import sglang.
Can someone help reducing the importing time?

```
> time python -c "from sglang.srt.managers.scheduler import Scheduler"
All deep_gemm operations loaded successfully!
W0916 03:22:46.348000 152559 torch/utils/cpp_extension.py:2425] TORCH_CUDA_ARCH_LIST is not set, all archs for visible cards are included for compilation.
W0916 03:22:46.348000 152559 torch/utils/cpp_extension.py:2425] If this is not desired, please set os.environ['TORCH_CUDA_ARCH_LIST'] to specific architectures.

real	0m12.310s
user	0m13.728s
sys	0m23.912s
```

**Comment by zhyncs** [comment]
I also encountered a similar issue on H200.

**Comment by ping1jing2** [comment]
I'm interested in this task, hope to solve it.

**Comment by vincent-shop** [comment]
Most of it is `import torch` and `openai` lib imports, along with a few others.
Also, the lazy import of anything from `transformers`, some PIL, type libs.
We should also have some kind of global lazy loader, but TBD the most effective way to implement this module.

**Comment by ping1jing2** [comment]
sync information(still don't find the solution...)

test on Ascend(aarch64) sever
- SGlang-main(import_schedule_in_sglang_main):
   it costs 11s ≈ 2s(vllm) + 5.5s(torch) + 1s(deepseekvl2 + transformers) + 0.7s(transformers) + 0.7s(managers)
- SGlang-main(without_import_vllm):
   it costs 11s ≈ 1.5s(compressed_tensors) + 5.9s(torch) + 1.3s(fastapi + openai) + 0.9s(transformers) + 0.7s(managers)
- vllm-main:
   it costs 10s ≈ 1s(model_loader) + 5.5s(torch) + 1.5s(layers.quantization) + 0.7s(transformers)

**import_schedule_in_sglang_main**
![Image](https://github.com/user-attachments/assets/7ac9d48f-58d0-4cb3-b013-fd1575f05c38)
**import_schedule_without_vllm**
![Image](https://github.com/user-attachments/assets/3050760e-6350-45ab-a759-00cc3c5700d0)
**import_llmEngine_in_vllm_main**
![Image](https://github.com/user-attachments/assets/617809a7-c78d-41be-bfe2-89b3a982a776)

**Comment by ping1jing2** [comment]
(I'm not familiar with time analysis on cpu, so i share this wonderful method)

how to reproduce these file?
1. get log file
   `time python -X importtime -c "from sglang.srt.managers.scheduler import Scheduler" 2> import_sglang.log`
2. parse log file on [python-importtime-graph](https://kmichel.github.io/python-importtime-graph/) website
3. download the .svg file

**Comment by LucasArray** [comment]
Hi @ping1jing2, are you still working on this? It has been quiet since October, so I would like to pick it up if you do not mind.

To avoid what happened to #12510 (a large refactor that stalled waiting for an approach review and then hit conflicts), my plan is small incremental PRs instead of one big one:

Re-profile on current main with python -X importtime (same method you shared above) to get an up-to-date breakdown.
Open a first small PR deferring only the top one or two safe-to-lazy imports, using the existing LazyImport utility, with before/after numbers in the description.
If that lands and the approach looks right, follow up with more small PRs, one module group at a time.

One scope question: since import torch dominates and cannot realistically be deferred for real server startup, is "reduce the non-torch import time" the right success criterion here?

If this sounds good, I'll start with the profiling and post the numbers back here.
